### PR TITLE
remove VERSION file

### PR DESCRIPTION
**Pull request by vieux**
removing the `VERSION` file since it doesn't make sense in the Moby context.
Docker CE version and branch management is now done in https://github.com/docker/docker-ce

See https://github.com/docker/docker-ce/blob/master/VERSION and https://github.com/docker/docker-ce/tree/17.06

I decided to go to an hardcoded "dev" version for now.

/cc @tianon @tiborvass @mlaventure @andrewhsu @thaJeztah

### CODE REVIEW

**Comment by thaJeztah** [comment]
LOL, was actually thinking of doing this today; can you also remove https://github.com/moby/moby/blob/master/poule.yml#L38-L42 ?

**Comment by vieux** [comment]
@thaJeztah PTAL

**Review by vdemeester — APPROVED** [review]
LGTM 👼

File: poule.yml
Comment (thaJeztah) [inline poule.yml]: while people still report docker issues, I think we should keep the `version-label`

**Review by thaJeztah — CHANGES_REQUESTED** [review]

File: poule.yml
Comment (vieux) [inline poule.yml]: oh I see

Comment (vieux) [inline poule.yml]: in the near future, ppl should probably not report issues in moby/moby, but I'll add back for now

**Review by thaJeztah — APPROVED** [review]
LGTM (if green)

next step; we need to decide how to handle API versioning

**Comment by thaJeztah** [comment]
Windows is failing - halp @johnstep

**Comment by jhowardmsft** [comment]
Was talking to @vieux on Slack to help, but I'm not sure I like the direction this is heading. Can we not have a moby version? And have docker do what-ever it needs to do to build with a docker version?

**Comment by vieux** [comment]
I think we're confusing to things, docker != moby therefore the docker version and moby version are different.

The moby version can be found here https://github.com/moby/tool/blob/master/cmd/moby/main.go#L16 (soon to me moved back to moby/moby)

The docker version can be found here https://github.com/docker/docker-ce/blob/master/VERSION

**Comment by AkihiroSuda** [comment]
CI seems failing

```
23:28:46 =================================== FAILURES ===================================
23:28:46 _________________________ InformationTest.test_version _________________________
23:28:46 /docker-py/tests/integration/api_client_test.py:19: in test_version
23:28:46     self.assertEqual(len(res['Version'].split('.')), 3)
23:28:46 E   AssertionError: 1 != 3
23:28:46 ======== 1 failed, 205 passed, 31 skipped, 2 xfailed in 281.74 seconds =========
```

File: hack/make.ps1
Comment (johnstep) [inline hack/make.ps1]: Update the message to reference Docker.windows instead of VERSION.

**Comment by thaJeztah** [comment]
We could leave a VERSION file and put, idk `0.1-dev` in it

**Comment by vieux** [comment]
@thaJeztah how about just "dev" I don't want to put a number since we have no plans to update it.

**Comment by thaJeztah** [comment]
@vieux my train of thought was that having 0.01-dev or 0.1-dev would potentially cause less issues with tools/code that's parsing a version string

**Comment by vieux** [comment]
@moby/moby-maintainers wdyt ? I'm still reluctant to use 0.1

**Comment by tiborvass** [comment]
I'm in favor for removing

**Comment by stevvooe** [comment]
Why don't we just remove the version file and use git tags?

**Comment by mlaventure** [comment]
@vieux I don't think that'd be an issue as those parsing tool are use against docker binaries. Moby not having releases, it shouldn't matter.

**Comment by AkihiroSuda** [comment]
What's the current status?

**Comment by vieux** [comment]
@mlaventure @stevvooe keep the file allows for an easy override from the docker-ce and docker-ee packaging.

**Comment by mlaventure** [comment]
@vieux the version string can be generated by the packing (using the repo own VERSION file), I think versioning should be left to the packaging.

As long as the engine Makefile allows for the ldflags to be augmented so we can override the value found in the `version.go` file, it shouldn't be an issue.

**Comment by mlaventure** [comment]
@andrewhsu WDYT ^^

**Comment by andrewhsu** [comment]
The docker/docker-ce repo is overriding the `VERSION` file for building packages for release: https://github.com/docker/docker-ce/pull/80/files

So, as long as there is some way for a user of the moby/moby repo to override the version string output of `docker --version`, I think any change would be acceptable.

**Comment by AkihiroSuda** [comment]
Note: There is a plan to start the version from 2.0
https://forums.mobyproject.org/t/topic-find-a-good-and-non-confusing-home-for-the-remaining-monolith/37/54

**Comment by mavenugo** [comment]
I heard few other 👍 to @thaJeztah's [comment](https://github.com/moby/moby/pull/33455#issuecomment-305826110)  to change the VERSION to `0.1-dev`.
cc @mlaventure @tiborvass @cpuguy83 @dnephin can you pls confirm with your suggestion ?

**Comment by dnephin** [comment]
As I see it there are two topics being discussed here:
1. What is the moby version?
2. How do we document the version?


**What is the moby version?**

I think the version should be 0.1. We've made the statement that this is a new effort, so we should start the versioning from the beginning again. I think v0.1 best reflects the changes that should be coming to this repo. We're talking about completely changing the architecture, splitting out major pieces, etc. 

** How do we document the version**

+1 to the suggestion from @tiborvass , @stevvooe, and @mlaventure to remove the file. I think we can use git tags and pass the version with ldflags

**Comment by cpuguy83** [comment]
I agree with removing. Consumers can even use a build flag to set the version if they want something compiled in.

**Comment by tiborvass** [comment]
@cpuguy83 @dnephin if there are any unknown issues with CIs expecting some version in master, maybe something like `0-dev` or `0.0-dev`. Ideally the default could just be `dev`.

**Comment by mlaventure** [comment]
@tiborvass CI can always be updated :)

I think it'll be a case of updating `hack/make.sh` and `hack/make.ps1` when they define the `VERSION` variable by reading out the content of the `VERSION` file

**Comment by dnephin** [comment]
Anything blocking this? Can we proceed with this change?

**Comment by AkihiroSuda** [comment]
ping @vieux @cpuguy83 @tiborvass @andrewhsu @dnephin

**Comment by stevvooe** [comment]
Wait, why aren't we removing the file?

We should be using tags for versioning. Let's remove the file and fix up the problems as they come.

**Comment by AkihiroSuda** [comment]
closing per new pr
 https://github.com/moby/moby/issues/35368

**Pull request closed without merging**
